luigi/task.py

```diff
@@ -61,6 +67,19 @@ class Task2(luigi.Task):
     Register._default_namespace = namespace
 
 
+def task_id_str(task_family, params):
```

Comment:
I don't want it private because the whole point is to expose it to the scheduler so it can apply correct ids to generated tasks. I'll add a docstring.